pkg/builder/openapi.go

```diff
@@ -45,7 +45,46 @@ type openAPI struct {
 
 // BuildOpenAPISpec builds OpenAPI spec given a list of webservices (containing routes) and common.Config to customize it.
 func BuildOpenAPISpec(webServices []*restful.WebService, config *common.Config) (*spec.Swagger, error) {
-	o := openAPI{
+	o := newOpenAPI(config)
+
+	err := o.init()
+	if err != nil {
+		return nil, err
+	}
+	err = o.buildPaths(webServices)
+	if err != nil {
+		return nil, err
+	}
+	err = o.finishSpec()
+	if err != nil {
+		return nil, err
+	}
+
+	return o.swagger, nil
+}
+
+// BuildOpenAPISpecForResource builds a partial OpenAPI spec given a sample object and common.Config to customize it.
+func BuildOpenAPISpecForResource(model interface{}, config *common.Config) (*spec.Swagger, error) {
```

Comment:
Maybe you should split into different PRs or at least commits when you're doing different things (refactor AND add new methods)

Reply:
It is actually not valid if it does not have paths.